CIP XXX Marlowe Oracle Protocol/v0.2.0.md

```diff
@@ -0,0 +1,268 @@
+
+This CIP describes a more flexible, transparent and composable oracle solution for Cardano, based on the Marlowe smart contract language, using the `Choice` construct in Marlowe to deliver the oracle value within a running smart contract, as well as making that value available to other Cardano other smart contacts, whether written in Marlowe, Aiken, Plutus or other languages.
+
+## Motivation: why is this CIP necessary?
+
+External data is used within the vast majority of smart contracts running on blockchain, and provision of this data by oracle providers is essential to these contracts running effectively and securely. The “push” model of data provides an ~80:20 solution: 80% of needs are satisfied by a limited set of data feeds, particularly those based on DeFi information of various kinds; it is, however, unable to service the “long tail” of potential demands for data of a more specialized kind, and to do this in practice a “pull” model of data provision is required.
+
+The mechanism outlined delivers data through the `Choice` construct in the Marlowe smart contract language. Executing an instance of this construct in a running contract delivers an external integer value that can be directly accessed within the contact, and also be made available for use by other smart contracts on Cardano.
+
+For this approach to work in practice, this CIP needs to address the following high-level questions:
+
+* How to identify what data is requested, and from which source, including an access mechanism for the source.
+* How results of oracle requests are rendered for use within Marlowe.
+* How results are made available on-chain to other Cardano smart contracts, including information about the data source, format and rendering.
+* How the results of particular instances of data requests are identified, both within and outside running Marlowe contracts.
+* The security model assumed by the oracle protocol.
+
+These high-level questions are answered in the next section, together with a discussion of implementation-related issues such as timeouts and error handling, and potential extensions of the protocol, such as access mechanisms.
+
+## Specification
+
+This section described the Marlowe Oracle Protocol in detail. Assumptions made in the description are listed here, before the specification itself.
+
+### Assumptions
+
+* No distinction is made between the data source and the oracle provider. In future versions of this protocol, this distinction may be possible.
+* IPFS is used as a standard for content-addressable off-chain data here; other equivalent mechanisms may be used.
```

Comment:
@simonjohnthompson I changed this line into fully open approach regarding the off-chain storage.